src/var.rs

```diff
@@ -13,7 +13,7 @@ use crate::{
 };
 
 /// The signature of a hint function
-pub type HintFn<B: Backend> = dyn Fn(&B, &mut WitnessEnv<B::Field, B::CellVar>) -> Result<B::Field>;
+pub type HintFn<B: Backend> = dyn Fn(&B, &mut WitnessEnv<B::Field>) -> Result<B::Field>;
```

Comment:
yeah. I am wondering if it is possible to do poseidon for the kimchi part without using closure? (all via arithmetic)

Reply:
the answer is no because kimchi is a special gate :D so we need hints there, but I think it's enough to have a Value::Poseidon that knows exactly what the hint is (as opposed to pass it as a closure). Can you create an issue for that? I think it'd be nice for someone else to tackle :)